nodejs/week1/src/backend/routes/meal.js

```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const dataAccess = require("../dataAccess.js");
+const router = express.Router();
+const meals = "/data/meals.json";
+const reviews = "/data/reviews.json";
+
+router.get("/meal", (request, response) => {
+  let mealsData = dataAccess.getDataFromJson(meals);
+  let reviewsData = dataAccess.getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  let randomMeal =
+    mealsWithReviews[Math.floor(Math.random() * mealsWithReviews.length)];
```

Comment:
This code (getting a random item from an array) is duplicated across several files. When you find yourself writing the same code more than twice, consider whether you can extract a helper function that you can reuse from both places. This principle is commonly known as DRY (Don't Repeat Yourself). Keeping your code DRY helps you right more maintainable code because:

* If there's a bug in this code, it's only in one place and you only need to fix it once
* When you create a helper function other team mates can benefit from it and reuse your code
* When someone looks at your code they need to read it slowly and understand what is going on. When they read a helper function like getRandomInteger(maxValue) they automatically understand what the function does without needing to exert their cognitive abilities a lot.